library/ctr_drbg.c

```diff
@@ -294,8 +295,17 @@ void mbedtls_ctr_drbg_update( mbedtls_ctr_drbg_context *ctx,
         if( add_len > MBEDTLS_CTR_DRBG_MAX_SEED_INPUT )
             add_len = MBEDTLS_CTR_DRBG_MAX_SEED_INPUT;
 
-        block_cipher_df( add_input, additional, add_len );
-        ctr_drbg_update_internal( ctx, add_input );
+        /*
+         * block_cipher_df() and ctr_drbg_update_internal() may return error codes.
+         * If they return error, abort the operation, since this public API
+         * does not have a return code, we can't return the error codes to the caller
+         */
+        ret = block_cipher_df( add_input, additional, add_len );
+        if( ret != 0)
+            return;
+        ret = ctr_drbg_update_internal( ctx, add_input );
+        if( ret != 0)
```

Comment:
Style: This might not even be worth pointing out, but there is a missing ` ` before `)`...